src/v/raft/recovery_throttle.h

```diff
@@ -135,11 +138,51 @@ class recovery_throttle {
         _rate = per_core_rate;
     }
 
+    void setup_metrics() {
+        setup_internal_metrics();
+        setup_public_metrics();
+    }
+
+    void setup_internal_metrics() {
+        if (config::shard_local_cfg().disable_metrics()) {
+            return;
+        }
+
+        // TODO: Delete this metric after public is supported in scraping
+        // configurations
+        namespace sm = ss::metrics;
+        _internal_metrics.add_group(
```

Comment:
nit: maybe we can have just a single function that takes a metrics endpoint as a parameter

Reply:
Didn't think about it
But I think that now it is easier to just delete this function without refactoring code later.